hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3971,11 +3972,139 @@ func (r *HostedClusterReconciler) validateHostedClusterSupport(hc *hyperv1.Hoste
 }
 
 func (r *HostedClusterReconciler) validateNetworks(hc *hyperv1.HostedCluster) error {
-	errs := validateSliceNetworkCIDRs(hc)
+	var errs field.ErrorList
+	errs = append(errs, validateNetworkStackAddresses(hc)...)
+	errs = append(errs, validateSliceNetworkCIDRs(hc)...)
+	errs = append(errs, validateAdvertiseAddress(hc)...)
 
 	return errs.ToAggregate()
 }
 
+// validateAdvertiseAddress perform validations over the AdvertiseAddress
+func validateAdvertiseAddress(hc *hyperv1.HostedCluster) field.ErrorList {
+	var errs field.ErrorList
+	errs = append(errs, checkAdvertiseAddressOverlapping(hc)...)
```

Comment:
this all can just be `return checkAdvertiseAddressOverlapping(hc)`.
Also all the upper level function does is call the inner one, so one could be removed.